src/main/java/org/roy/loadx/transaction/TransactionPrinter.java

```diff
@@ -19,19 +22,38 @@ private static void format(String format, Object... args) {
 	}
 
 	public void print() {
-		printFormatted("Transaction", "Average", "Count");
-		printFormatted("===========", "=======", "=====");
+		List<String[]> output = new ArrayList<>();
+		output.add(new String[] {"Transaction", "Average", "Count"});
+		int maxTxLength = 11; // length of 'Transaction'
+		
 		for (String name : transactionAggregator.getSortedTransactionNames()) {
 			TransactionData transactionData = transactionAggregator.getTransactionData(name);
 			long roundedAverage = Math.round(transactionData.getAverageDurationMilli());
 			String average = String.valueOf(roundedAverage);
 			String count = String.valueOf(transactionData.getTransactionCount());
-			printFormatted(name, average, count);
+			output.add(new String[] {name, average, count});
+			if(name.length() > maxTxLength)
+				maxTxLength = name.length();
 		}
-		println();
+		printFormatted(output, maxTxLength);
 	}
 
-	private void printFormatted(String name, String count, String average) {
-		format("%-15s | %10s | %10s", name, count, average);
+	private void printFormatted(final List<String[]> data, final int maxTxLength) {
```

Comment:
use better identifier name than data, use proper class instance of List<String[]>, full name for maxTxLength